src/app/screens/Home/index.tsx

```diff
@@ -242,31 +245,54 @@ function Home() {
               onClick={async () => {
                 try {
                   setLoadingSendSats(true);
-                  const origin = {
+
+                  const originData = {
                     external: true,
                     name: lnData[0].name,
                     host: lnData[0].host,
                     description: lnData[0].description,
                     icon: lnData[0].icon,
                   };
+
                   if (lnData[0].method === "lnurl") {
-                    navigate(
-                      `/lnurlPay?lnurl=${
-                        lnData[0].address
-                      }&origin=${encodeURIComponent(JSON.stringify(origin))}`
-                    );
+                    const lnurl = lnData[0].address;
+                    const lnurlDetails = await lnurlLib.getDetails(lnurl); // throws if invalid.
+
+                    if (lnurlDetails.tag === "payRequest") {
+                      // rather pick the key/value the component needs
+                      // the data coming from `getDetails` contains deep nested non-serializable values which results in a `null` navigation state
```

Comment:
the non-serializable value is the URL here: https://github.com/getAlby/lightning-browser-extension/blob/master/src/common/lib/lnurl.ts#L82
I think we don't need that and could simple pass keep the URL as string and init a URL object where needed. 

it could even be the case that this `url` value is currently not even accessed.

Reply:
Thanks for the info. Adjusted this 👍